framework/src/utils/InputParameterWarehouse.C

```diff
@@ -17,7 +17,10 @@
 
 InputParameterWarehouse::InputParameterWarehouse() :
     Warehouse<InputParameters>(),
-    _name_to_shared_pointer(libMesh::n_threads())
+    _input_parameters(libMesh::n_threads()),
+    _system_to_index(libMesh::n_threads()),
+    _syntax_to_index(libMesh::n_threads()),
+    _tag_to_index(libMesh::n_threads())
```

Comment:
No, removing threads will not be trival because of the way I am constructiong objects. 
`Factory::create` is what injects InputParameters into the warehouse, which provides back a reference that is used to build the actual object.  

`Factory::create` is inside loops that creates n-thread objects. So, to remove this will require that the reference to the InputParameters is setup, then the passed into Factory::create on each of the n-thread objects. I am not sure the best way to do this at the moment. Perhaps Factory::create always returns a vector of objects and can create threaded copies...I am not sure. I don't want to tackle that at the moment.